crates/sui-core/src/authority_active/checkpoint_driver/mod.rs

```diff
@@ -0,0 +1,551 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    collections::{BTreeMap, BTreeSet},
+    sync::Arc,
+    time::Duration,
+};
+
+use parking_lot::Mutex;
+use sui_types::{
+    base_types::AuthorityName,
+    error::SuiError,
+    messages_checkpoint::{
+        AuthenticatedCheckpoint, AuthorityCheckpointInfo, CertifiedCheckpoint, CheckpointContents,
+        CheckpointRequest, CheckpointResponse, CheckpointSequenceNumber, SignedCheckpoint,
+        SignedCheckpointProposal,
+    },
+};
+
+use crate::{
+    authority_aggregator::ReduceOutput,
+    authority_client::AuthorityAPI,
+    checkpoints::{proposal::CheckpointProposal, CheckpointStore},
+};
+
+#[cfg(test)]
+pub(crate) mod tests;
+
+use super::ActiveAuthority;
+
+pub async fn checkpoint_process<A>(_active_authority: &ActiveAuthority<A>)
+where
+    A: AuthorityAPI + Send + Sync + 'static + Clone,
+{
+    println!("START ACTIVE CHECKPOINT PROCESS!");
+    if _active_authority.state._checkpoints.is_none() {
+        // If the checkpointing database is not present, do not
+        // operate the active checkpointing logic.
+        return;
+    }
+
+    // Safe to unwrap due to check above
+    let state_checkpoints = _active_authority
+        .state
+        ._checkpoints
+        .as_ref()
+        .unwrap()
+        .clone();
+
+    tokio::time::sleep(Duration::from_millis(1220)).await;
+
+    loop {
+        // First, get the latest summaries and proposals
+        let (checkpoint, proposals) = get_latest_proposal_and_checkpoint_from_all(
+            _active_authority,
+            Duration::from_millis(200),
```

Comment:
What is this timer? Should we have a Const instead of `200ms` to make the name of the constant explicit?